### PR TITLE
OSDEV-3223: [Infrastructure] Upgrade deprecated Python Lambda runtimes (3.8 and 3.10) to a supported version

### PR DESCRIPTION
**Pull request by valencia-max**
## Jira Ticket

[OSDEV-3223](https://opensupplyhub.atlassian.net/browse/OSDEV-3223)

---

## Summary of Changes

Upgraded all Python Lambda functions off the deprecated 3.8/3.10 runtimes to 3.13 (3.10 loses AWS support Oct 31, 2026; 3.8 already lost support in 2024). Changed the `runtime` declaration in Terraform for all 7 affected functions (`alert_batch_failures`, `alert_sfn_failures`, and the 4 ContriBot Lambdas on 3.8/3.10, plus `nlb_targets_registrar`), updated `contribot_process_list`'s Makefile cross-compile target to 3.13, and bumped the ContriBot CI test job and `pyrightconfig.json` to match. Also fixed 4 regex string literals in `lib/contribot.py` that raised `SyntaxWarning` on 3.12+ (unescaped `\s`/`\.` in non-raw strings) — no behavior change, just silences the warning.

Target runtime (3.13, not the newer 3.14) was chosen because no `pandas` wheel exists yet for 3.14.

Aditionally:
`func…AlertBatchFailures` and `func…AlertStepFunctionsFailures` were failing **every** invocation with `Runtime.ImportModuleError: No module named 'alert_sfn_failures'`. Three faults compounded:

1. The committed archives were built by hand from the wrong directory in 2023 — each held `Makefile`, `requirements.txt` and `src/<name>.py`, putting the handler module one level below the archive root where `<name>.handler` cannot find it.
2. They bundled no dependencies, so `import rollbar` would have failed next.
3. **`source_code_hash` was commented out** on both resources. Each function's `Makefile` has always produced the correct layout, and `deployment/infra plan` has run it on every deploy since — but with nothing describing the archive contents, a rebuilt zip was never a diff and `apply` never re-uploaded. The 2023 bytes stayed deployed for three years.

Fixed by enabling `source_code_hash` on both, gitignoring the archives (a zip is a build artifact; committing one is what let these rot), and generalising `deployment/infra`: `CONTRIBOT_LAMBDAS` → `LAMBDA_ARCHIVES` with the two alert functions added. That removes the special-cased `make` calls and gives them the same build-at-plan → S3 → byte-identical-at-apply handoff, plus the build-before-destroy that `filebase64sha256()` now requires. The Lambda@Edge archives are deliberately excluded — no `Makefile`, so they stay committed.

---

## Type of Change

- [x] Backend change (API, database, business logic)
- [x] Architectural change (affects structure, contracts, or shared modules)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] UI / frontend change
- [ ] Refactor (no functional change)
- [ ] Breaking change (existing functionality may not work as before)
- [ ] Documentation / tooling only

---

## Testing

All 174 existing ContriBot unit tests pass on Python 3.13.13. Verified `contribot_process_list`'s Makefile builds a working zip against 3.13 (correct `cpython-313` `.so` files) and that `numpy`/`pandas`/`rapidfuzz`/`googleapiclient` and the handler import cleanly from it. Verified `alert_batch_failures`, `alert_sfn_failures`, and `register_nlb_targets` import cleanly on 3.13. Not yet run: deployment to AWS or invocation of the upgraded functions in a real environment.

---

## Known TODO Items

AC verification requiring AWS access is not done in this PR: confirming `list-functions` returns empty for python3.8/3.10 across all regions (including published versions), invoking each function in production with a representative payload, and confirming the AWS Health notifications clear.

---

### Implementation

- [x] My code follows the project's style guidelines and naming conventions.
- [x] I have removed debug logs, commented-out code, and unused imports.
- [x] Any AI-assisted code (e.g., Claude) has been reviewed, understood, and adapted to fit our codebase.
- [x] I have considered backward compatibility and migrations where applicable.
- [x] The diff contains no secrets, internal document/spreadsheet/folder IDs, internal board or channel identifiers, internal policy thresholds, or staff contact details.

### Testing & Validation

- [ ] I have manually tested the change in a local/dev environment.
- [ ] I have added or updated unit tests for the new/changed behavior.
- [x] All existing tests pass locally.
- [ ] For UI changes: N/A
- [x] For backend changes: I have validated API contracts, error handling, and edge cases.
- [x] For architectural changes: I have documented the design decision and notified affected teams.

### Documentation & Communication

- [x] I have updated relevant documentation (README, ADRs, inline comments, API docs).
- [ ] I have updated the Jira ticket status and added any relevant notes.

### Final Review

- [x] I have performed a self-review on my PR and I am presenting my best work for my peers to review.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[OSDEV-3223]: https://opensupplyhub.atlassian.net/browse/OSDEV-3223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: deployment/terraform/contribot_lambda.tf [hunk 1]
```diff
@@ -159,7 +159,7 @@ resource "aws_lambda_function" "contribot_process_list" {
   description      = "ContriBot task that processes a single facility list."
   role             = aws_iam_role.contribot_lambda.arn
   handler          = "handler.handler"
-  runtime          = "python3.10"
+  runtime          = "python3.13"
```
[the remark this answers is not in the document]
Comment (valencia-max) [reply deployment/terraform/contribot_lambda.tf:162]: Done

**Review by daniellekain — DISMISSED** [review]

**Review by daniellekain — APPROVED** [review]
